draft/features-roadmap.rst

```diff
@@ -0,0 +1,116 @@
+=========================
+CEP XXXX: Feature Roadmap
+=========================
+
+:CEP: XXXX
+:Author: Omer Katz
+:Implementation Team: Omer Katz
+:Shepherd: Omer Katz
+:Status: Draft
+:Type: Informational
+:Created: 2020-08-25
+:Last-Modified: 2020-08-25
+
+.. contents:: Table of Contents
+   :depth: 3
+   :local:
+
+.. role:: breaking-change
+.. role:: new-feature
+
+Abstract
+========
+
+This CEP details the roadmap for planned new features and breaking changes for Celery NextGen.
+The description of these features is done in other CEPs. If one is available, we will provide a link to the matching CEP.
+Otherwise, we will link to an issue which describes the motivation behind the feature or breaking change.
+
+The document will be updated from time to time whenever we intend to release new major versions of Celery.
+
+Specification
+=============
+
+Celery 5.x
+-----------
+
+Celery 5.0 is the first version to only support Python 3.
+
+Breaking Changes
+~~~~~~~~~~~~~~~~
+
+- :breaking-change:`Drop support for Python 2.7.`
+- :breaking-change:`Drop support for Python 3.5.`
+- :breaking-change:`Drop support for the Riak result backend.`
+- :breaking-change:`Remove the deprecated old tasks API.`
+
+New Features
+~~~~~~~~~~~~
+
+- :new-feature:`New CLI based on Click.`
+
+Celery 6.x
+-----------
+
+Celery 6.0 will introduce a new event loop, drop support for the prefork, gevent & eventlet worker pools
+and will allow executing tasks defined with `async def` among other features.
+
+Support for a single consumer and multiple workers will be reinstated in 7.x.
+
+Breaking Changes
+~~~~~~~~~~~~~~~~
+
+- :breaking-change:`Drop support for the prefork workers pool.`
```

Comment:
can we add deprecation warning for this worker pools? in celery 5.1/.5.2

Reply:
No. People complained about us dropping features in minor releases.
We should stick to SemVer even if it is less flexible.

Reply:
OK